src/config.rs

```diff
@@ -211,10 +213,34 @@ impl ConfigHelpItem {
     }
 }
 
+/// This is used by Config to track which config parameters are accessed during
+/// formatting. It uses a RefCell for interior mutability, as we don't want to
+/// require a mutable reference to Config in order to access configuration.
+#[derive(Clone, Default)]
+struct ConfigTracker {
```

Comment:
I've done this. I think it has simplified the code too, as I now have a single `Cell<bool>` for each config item to track whether it's been accessed, instead of a `HashSet<String>`.